docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,595 @@
+# OKEP-4368: No-overlay Mode For Layer-3 Networks
+
+* Issue: [#5259](https://github.com/ovn-kubernetes/ovn-kubernetes/issues/5259)
+* Authors: Riccardo Ravaioli, Peng Liu
+
+## Problem Statement
+
+Currently OVN-Kubernetes uses Geneve as its transport method on the overlay
+network for east/west traffic; this adds overhead and reduces throughput. By
+leveraging ovn-kubernetes support for BGP, we want to provide a way for users to
+enable a no-overlay mode, which would disable Geneve encapsulation and use
+direct routing between nodes for east/west traffic on selected networks.
+
+Many environments, particularly on-premise deployments or those with dedicated
+networking infrastructure, prefer to utilize the underlying physical network's
+routing capabilities directly. This "no-overlay" approach can offer several
+benefits:
+
+* Improved Performance: Eliminates encapsulation/decapsulation overhead,
+  potentially leading to lower latency and higher throughput for inter-pod
+  communication.
+* Simplified Troubleshooting: Traffic paths are more transparent as they align
+  with the physical network's routing tables, simplifying debugging and network
+  visibility.
+* Leverage Existing Network Infrastructure: Integrates more seamlessly with
+  existing BGP-capable network devices, allowing for direct routing to pod IPs.
+* Reduced Resource Consumption: Less CPU cycles spent on
+  encapsulation/decapsulation.
+
+## Goals
+
+* Support no-overlay mode for the default network
+* Support no-overlay mode for layer-3 Cluster User Defined Networks (CUDNs) in
+  the VRF-lite mode .
+* A cluster can have networks operating in overlay and no-overlay modes
+  simultaneously.
+* Use the route advertisements feature to exchange routes to node subnets across
+  the cluster.
+* Allow direct pod-to-pod communication without any overlay transport for
+  West-East traffic.
+* Ensure East-West network policies are still enforced by OVN.
+* Maintain compatibility with existing OVN-Kubernetes features where applicable
+  (e.g., Services, NetworkPolicy, EgressIP, EgressFirewall, AdminNetworkPolicy).
+* Compatible with both local gateway and shared gateway modes.
+  
+## Future Goals
+
+* When OVN-Kubernetes supports BGP for UDNs, extend the UDN API to enable
+  no-overlay mode on the selected network
+* Support toggling no-overlay mode on/off for an existing network.
```

Comment:
Support togging no-overlay mode on/off runtime by patch the CUDN resource, right?

Reply:
The default network and UDNs. We can remove the API restriction if this use case makes sense.